tests/istanbul/eip152_blake2/test_blake2.py

```diff
@@ -0,0 +1,501 @@
+                data_1="0x94e963dcaa85d33dad6c0043b6700f5e227a2d8bed804bd16970e64fa6f1e163",
+                data_2="0x07399239bcddf968612c8c9ba953d9b173575c31ef269c3a8721cb9bf1c02012",
+            ),
+            id="EIP-192-0128-16",
+        ),
+        pytest.param(
+            Blake2Input(rounds=b"\x01\x00", data=b"\x10"),
+            ExpectedOutput(
+                call_return_code="0x01",
+                data_1="0x7476f1f8e159b30b156b0e9fffbaee5badbb45abb488ea9cfa04f60d3a096408",
+                data_2="0x7535a649e438ba993f03cfc0d8d8676a792030a996b6a6fde5c29108b6bfb871",
+            ),
+            id="EIP-192-0256-16",
+        ),
+        pytest.param(
+            Blake2Input(rounds=b"\x02\x00", data=b"\x10"),
+            ExpectedOutput(
+                call_return_code="0x01",
+                data_1="0xa65aadbf393aad57c4b06d6471134c5c01002c23dfe8290e115e024e05bc1bf1",
+                data_2="0x084d1651de54a83902ed582cb8f2ba381c69687cceaecea0cd8fe5529f86686e",
+            ),
+            id="EIP-192-0512-16",
+        ),
+        pytest.param(
+            Blake2Input(rounds=b"\x04\x00", data=b"\x10"),
+            ExpectedOutput(
+                call_return_code="0x01",
+                data_1="0xbc112be5618b20d24be64c9e1c6efd63fea38cc79d53692fad6568b16e953eb6",
+                data_2="0x128c1ec8ffaf9a2d69e3cb043d6e11e1c7afd48573311052b6e7ec0960371186",
+            ),
+            id="EIP-192-1024-16",
+        ),
+        pytest.param(
+            Blake2Input(rounds=b"\x00\x10", data=b"\x00x"),
+            ExpectedOutput(
+                call_return_code="0x01",
+                data_1="0x7df6f69476a03ae29e944814846460b058d1762fffe77f938ea723d1033de0d5",
+                data_2="0xbb1f8234bd73afaf955622fa2cdde95594577a8d53191908eb69b316a53c985b",
+            ),
+            id="EIP-192-0016-120",
+        ),
+        pytest.param(
+            Blake2Input(rounds=b"\x00\x20", data=b"\x00x"),
+            ExpectedOutput(
+                call_return_code="0x01",
+                data_1="0x38dfb111db6c59c39c9bea26c8c872620d89dec22fd7da93c47d0708a3973f52",
+                data_2="0x2858fd9c60fae53f366d7e2820040a8662b336de6d859764f20747acbb8999fe",
+            ),
+            id="EIP-192-0032-120",
+        ),
+        pytest.param(
+            Blake2Input(rounds=b"\x00\x40", data=b"\x00x"),
+            ExpectedOutput(
+                call_return_code="0x01",
+                data_1="0x59c7c9896cdc9fda0a77bd41adba14bdd6cb47cbbd7c338482f5382d7be16ac4",
+                data_2="0x4a2ddfe6833bf9a15737dd66469d2d0495d39a9cb3c8ed93152684d92a74f8bc",
+            ),
+            id="EIP-192-0064-120",
+        ),
+    ],
+)
+def test_blake2(
+    state_test: StateTestFiller,
+    yul: YulCompiler,
+    data: Blake2Input,
+    output: ExpectedOutput,
+    pre: Alloc,
+):
+    """Test BLAKE2 precompile."""
+    env = Environment()
+
+    account = pre.deploy_contract(
+        code=yul(
```

Comment:
the end goal is to get rid of yul as it is external dependency and the compiled code might be altered by solc.
rewrite the code using Op.  (see in existing tests)